### PR TITLE
feat(inputs.cgroups): dont abort on first error, stop error message spam

### LINKED ISSUE #12338: inputs, cgroup: spams errors and exits collection too early on unknown file formats
### Relevant telegraf.conf

```toml
[[inputs.cgroup]]
  paths = ["/sys/fs/cgroup/my-group.slice/my-service.service/*"]
```

### Logs from Telegraf

```text
2022-12-06T10:09:55Z E! [inputs.cgroup] Error in plugin: /sys/fs/cgroup/my-group.slice/my-service.service/memory.max: unknown file format
```

### System info

Telegraf 1.23.4

### Docker

_No response_

### Steps to reproduce

1. Use cgroups v2
2. Use default memory.max for some service
3.  Point telegraf at this service
4. Observe errors

### Expected behavior

I would like for telegraf to inform me about the error once and not on every collection cycle.  Also telegraf should collect metrics from all other files in the directory best-effort even when collecting from the first file errors.

### Actual behavior

Telegraf errors out on the first error and skips the remaining files; also it spams this error on every collection. 

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by mhoffm-aiven**
# Required for all PRs

- [-] Updated associated README.md.
- [X] Wrote appropriate unit tests.
- [X] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #12338 

Added a small lookup table for files that we already reported errors on so that we dont report them on every `Gather` round. Also fixed a bug where a formatting error would lead to subsequent files not being collected.

### CODE REVIEW

File: plugins/inputs/cgroup/cgroup_test.go [hunk 1]
```diff
@@ -332,3 +332,32 @@ func TestCgroupStatistics_7(t *testing.T) {
...
+	require.Error(t, err)
+	require.Len(t, cg8.logged, 1)
+
+	tags := map[string]string{
+		"path": "testdata/broken",
+	}
+	fields := map[string]interface{}{
+		"memory.limit_in_bytes": int64(1),
+	}
+	acc.AssertContainsTaggedFields(t, "cgroup", fields, tags)
```
Comment (srebhan) [inline plugins/inputs/cgroup/cgroup_test.go:355]: You should use the Telegraf helpers for this
```suggestion
	require.Len(t, cg8.logged, 1)

	expected := []telegraf.Metric{
		metric.New(
			"cgroup",
			map[string]string{"path": "testdata/broken"},
			map[string]interface{}{"memory.limit_in_bytes": int64(1)},
			time.Unix(0, 0)
		},
	}
	testutil.RequireMetricsEqual(t, expected, acc.GetTelegrafMetrics(), testutil.IgnoreTime())
```

**Review by srebhan — COMMENTED** [review]
Thanks for taking care of this issue @mhoffm-aiven! I do have some minor comments in the code...

File: plugins/inputs/cgroup/cgroup_test.go (hunk 1 above)
Comment (mhoffm-aiven) [reply plugins/inputs/cgroup/cgroup_test.go:355]: :heavy_check_mark:

File: plugins/inputs/cgroup/cgroup_test.go [hunk 2]
```diff
@@ -332,3 +332,32 @@ func TestCgroupStatistics_7(t *testing.T) {
 	}
 	acc.AssertContainsTaggedFields(t, "cgroup", fields, tags)
 }
+
+var cg8 = &CGroup{
+	Paths:  []string{"testdata/broken"},
+	Files:  []string{"malformed.file", "memory.limit_in_bytes"},
+	logged: make(map[string]bool),
+}
+
```
[the remark this answers is not in the document]
Comment (mhoffm-aiven) [reply plugins/inputs/cgroup/cgroup_test.go:341]: :heavy_check_mark:

File: plugins/inputs/cgroup/cgroup.go [hunk 3]
```diff
@@ -14,12 +14,14 @@ var sampleConfig string
...
+	logged map[string]bool
 }
 
 func (*CGroup) SampleConfig() string {
 	return sampleConfig
 }
 
 func init() {
-	inputs.Add("cgroup", func() telegraf.Input { return &CGroup{} })
+	inputs.Add("cgroup", func() telegraf.Input { return &CGroup{logged: make(map[string]bool)} })
```
[the remark this answers is not in the document]
Comment (mhoffm-aiven) [reply plugins/inputs/cgroup/cgroup.go]: :heavy_check_mark:

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for also migrating the existing tests @mhoffm-aiven!

**Review by powersj — APPROVED** [review]
Thank you!

**Pull request merged**